Challenges/GoDownTheRabbitHole/Solutions/RioMH.js

```diff
@@ -0,0 +1,59 @@
+let parentsCache = [];
+
+function countParents(parentId, data) {
+    let cached = parentsCache[parentId];
+    let count = 1;
+
+    if (cached) {
+        count = cached;
+    } else {
+        for (let index = 1; index < data.length; index++) {
+            let item = data[index]
+            if (item.itemId == parentId && item.parentId != null) {
+                count += countParents(item.parentId, data);
+                break;
+            }
+        }
+    }
+
+    parentsCache[parentId] = count;
+    return count;
+}
+
+function getIndent(item, data) {
+    let indent = "";
+
+    if (item.parentId != 0) {
+        let repeatCount = countParents(item.parentId, data);
+
+        if (repeatCount > 0) {
```

Comment:
repeatCount seems to always be greater than 0. It's initialised with 1 on line 5.